docs/docs/03-language-reference.md

```diff
@@ -92,7 +92,7 @@ These don't affect the value of the number or how they are printed:
 | Name          | Extra information                               |
 | ------------- | ----------------------------------------------- |
 | `Array<T>`    | variable size array of a certain type           |
-| `Map<T>`      | map type (key-value with string keys)           |
+| `Map<T>`      | map type (key-value with string keys, keys may be any expression evaluating to a string)|
```

Comment:
IMO the syntax feels off for a couple reasons. For one, maps look similar to json objects but have widely different semantics. I feel like that will throw people off. This is more a point about how I think we should rid of the special map syntax, but if not we should at least avoid surprises:

```
let a = "stuff";
{ a => 2, b => 2 };
{ a: 2, b: 2 };
```

If we ever add punning, this incongruity would get even worse.


Additionally, allowing arbitrary expressions is way too much freedom in something that should be simple.
I can understand wanting string interpolation here, which already would allow for arbitrary expressions, but at least it signals to the user that they should keep their expression brief (like they are probably doing in all interpolations)

Reply:
I might be nitpicking but the above example should be:
```wing
let a = "stuff";
{ a => 2, "b" => 2 };
{ a: 2, b: 2 };
```
I think the motivation for moving to `=>` was to better differentiate between the two.
Anyway this seems like a language design issue. @eladb WDYT?

Reply:
> I might be nitpicking but the above example should be:

The fact that this mistake was so easy kinda proves the point (or I'm just dumb)

I'm fine with this PR going through and moving the bigger discussion to another time because I think we should just remove the map (and set) literals anyway. Maps are not important enough to have their own literal when we can support implicit conversion from Json -> Map or explicit initialization ala structs:

```
let a = Map<num> { a: 1 } // This doesn't work right now, but it could
let b: Map<num> = { a: 1 } // This already works
```

Reply:
I see, so part of that future discussion can be how do we provide a good syntax for expression json object keys (better than `{ [ expr ]: value}`, maybe just support string interpotations in keys is good enough for json.